### PR TITLE
Retry & Async

### PR DESCRIPTION
**Pull request by songshenzong**
##### You need to complete

- [x] unit tests and/or feature tests
- [x] documentation is changed or added

### CODE REVIEW

File: src/Filter/ClientFilter.php [hunk 1]
```diff
@@ -64,6 +64,32 @@ public static function clientName($clientName)
...
+    public static function retry($times)
+    {
+        if (!is_int($times)) {
+            throw new ClientException(
+                'Retry must be a int',
+                SDK::INVALID_ARGUMENT
+            );
+        }
+
+        if ($times === '') {
```
Comment (JacksonTian) [inline src/Filter/ClientFilter.php]: 空字符串是不是在上面就判断掉了啊

File: src/Filter/ClientFilter.php [hunk 2]
```diff
@@ -64,6 +64,32 @@ public static function clientName($clientName)
...
+            throw new ClientException(
+                'Retry cannot be empty',
+                SDK::INVALID_ARGUMENT
+            );
+        }
+
+        return strtolower($times);
```
Comment (JacksonTian) [inline src/Filter/ClientFilter.php]: 为啥要用 strtolower 呢。

File: src/Filter/ClientFilter.php (hunk 1 above)
Comment (songshenzong) [reply src/Filter/ClientFilter.php]: 对

File: src/Filter/ClientFilter.php (hunk 2 above)
Comment (songshenzong) [reply src/Filter/ClientFilter.php]: 我错了

**Pull request merged**